langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java

```diff
@@ -306,25 +316,34 @@ private Map<String, Object> prefillMetadata() {
     }
 
     private EmbeddingMatch<TextSegment> toEmbeddingMatch(Map<String, ?> item) {
-        Map<String, ?> additional = (Map<String, ?>) item.get(ADDITIONALS);
-        final Metadata metadata = new Metadata();
-        if (item.get(METADATA) != null && item.get(METADATA) instanceof Map) {
-            Map<String, ?> resultingMetadata = (Map<String, ?>) item.get(METADATA);
-            for (Map.Entry<String, ?> entry : resultingMetadata.entrySet()) {
-                if (entry.getValue() != null && !NULL_VALUE.equals(entry.getValue())) {
-                    metadata.add(entry.getKey(), entry.getValue());
-                }
-            }
+      Map<String, ?> additional = (Map<String, ?>) item.get(ADDITIONALS);
+      final Metadata metadata = new Metadata();
+      Map<String, ?> metadataMap = new HashMap<>();
+      if (metadataFieldName.isEmpty()) {
+        metadataMap = new HashMap<>(item);
+        // Remove text field from metadata if we store metadata in the root of the object
+        metadataMap.remove(textFieldName);
```

Comment:
what about `ADDITIONALS`? any other fields?

Reply:
I  think it is just ADDITIONALS, is a good point.

For the most part, I would expect folks to only use this in combination with `metadataKeys` honestly but did not want to return too much extra data if not.

Reply:
Also done for ADDITIONALS